### ISSUE TITLE
Cannot apply to buildSrc

### ISSUE THREAD

**Issue by pkubowicz** [JLLeitschuh/ktlint-gradle#743]
When I add the plugin to buildSrc/build.gradle.kts:

```kotlin
plugins {
    `kotlin-dsl`
    id("org.jlleitschuh.gradle.ktlint") version "12.0.3"
}

configure<org.jlleitschuh.gradle.ktlint.KtlintExtension> {
    version.set("1.1.1")
    filter {
        include("**/*.kts")
        exclude { element -> element.file.path.contains("/generated-sources/") }
    }
}

```

it fails on classes generated by Gradle, even though I explicitly instruct the plugin not to include them.

```
buildSrc/build/generated-sources/kotlin-dsl-external-plugin-spec-builders/kotlin/gradle/kotlin/dsl/plugins/_bf0f0d1de34ae2a280c4b3e0ab359712/PluginSpecBuilders.kt:32:9 Package name must not contain underscore (cannot be auto-corrected) (standard:package-name)
```

**Comment by JLLeitschuh** [comment]
Can you share a build scan?

Also, a minor reproducer would be nice.

@wakingrufus we could probably automatically filter out this directory when the kotlin-dsl plugin is applied. Thoughts?

**Comment by pkubowicz** [comment]
The scan is here: https://scans.gradle.com/s/enzehbyek47qs

Attached a simple reproducer.
[gradle-kotlin-buildsrc.zip](https://github.com/JLLeitschuh/ktlint-gradle/files/14020735/gradle-kotlin-buildsrc.zip)

**Comment by wakingrufus** [comment]
> Can you share a build scan?
> 
> Also, a minor reproducer would be nice.
> 
> @wakingrufus we could probably automatically filter out this directory when the kotlin-dsl plugin is applied. Thoughts?

We could. I don't see this problem in regular projects using the kotlin-dsl plugin, so what is different about buildSrc that causes this?

**Comment by xenomachina** [comment]
tl;dr: It looks like something is getting cached in `buildSrc/.gradle/buildOutputCleanup/` that's preventing changes to excludes from being honored.

---

I ran into this same problem: ktlint was reporting errors in the Kotlin code that Gradle generates in `buildSrc/`. I added an exclude like this...

``` kotlin
ktlint {
    filter {
        exclude {
            it.file.relativeTo(projectDir).startsWith(File("build"))
        }
    }
}
```

...and it continued reporting these errors. Eventually I used `git clean` to delete all ignored files from my repo, and then it started working.

I was able to narrow the problem down to `buildSrc/.gradle/buildOutputCleanup/` by conducting the following series of tests:

- Start from a clean repo, with the above exclude: WORKS
- Remove exclude: FAILS
- Add exclude back: FAILS
- `rm -rf buildSrc/.gradle/`: WORKS
- Remove exclude: FAILS
- Add exclude back: FAILS
- `rm -rf buildSrc/.gradle/kotlin/`: FAILS
- `rm -rf buildSrc/.gradle/buildOutputCleanup/`: WORKS
- Remove exclude: FAILS
- Add exclude back: FAILS
- `rm -rf buildSrc/.gradle/buildOutputCleanup/`: WORKS

"WORKS" and "FAILS" indicate the outcome of running `./gradlew :buildSrc:ktlintCheck` after the corresponding operation.
